app/models/floating_ip.rb

```diff
@@ -7,18 +7,14 @@ class FloatingIp < ApplicationRecord
   # TODO(lsmola) NetworkManager, once all providers use network manager rename this to
   # "ManageIQ::Providers::NetworkManager"
   belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::BaseManager"
-  # TODO(lsmola) NetworkManager, remove when all providers share the new network architecture
+  # Not going through network_port because of Amazon, old EC2 way allows to associate public Ip to instance, without
+  # any network_port used
```

Comment:
I'll see what I can do when I rewrite all providers. I started a bit of unification (creating fake network_port) but I don't want to get rid of this until I check out rest of the cloud providers.

In amazon you can have basically 2 paths vm -> floating_ip and vm -> network_port -> floating_ip, where the first one is some old EC2 way.

I can't simply solve this only on amazon side, since this is used e.g. in tables, that means the same SQL query has to be applicable over all providers.